app/models/language_attribute.rb

```diff
@@ -3,5 +3,5 @@
 class LanguageAttribute < BaseAttribute
   belongs_to :language
 
-  validates :resource, presence: true, uniqueness: {scope: [:resource, :key]}
+  validates :resource, presence: true, uniqueness: {scope: [:resource, :language, :key]}
```

Comment:
I know you're not adding it here, but seeing this made me question why the scope is on `:resource` when that is the relationship. With your change, now this validation is doing a query like:
```sql
SELECT 1 AS one 
FROM "language_attributes" 
WHERE "language_attributes"."resource_id" = $1 
AND "language_attributes"."resource_id" = $2 
AND "language_attributes"."language_id" IS NULL 
AND "language_attributes"."key" IS NULL LIMIT $3
[["resource_id", 1], ["resource_id", 1], ["LIMIT", 1]]
```
which obviously is redundant. I think you should clean it up while you're in here:
```suggestion
  validates :resource, presence: true, uniqueness: {scope: [:language, :key]}
```